arch/cortex-m/samv71-hal/src/hal.rs

```diff
@@ -244,10 +250,15 @@ impl SystemHal for Hal {
         peripherals.watchdog.feed();
     }
 
+    /// Enters critical section by disabling global interrupts.
     fn enter_critical() {
         cortex_m::interrupt::disable();
     }
 
+    /// Exits critical section by enabling global interrupts.
+    ///
+    /// # Safety:
+    /// This function should never be called in critical sections created with [`SystemHal::execute_critical`]
```

Comment:
Maybe use this new `warning` tag that is now available in the Rust doc?

Reply:
done